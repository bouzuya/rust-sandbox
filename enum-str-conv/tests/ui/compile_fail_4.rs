fn main() {
    fn unknown_fn(_s: String) {
        unreachable!()
    }

    #[derive(enum_str_conv::EnumStrConv)]
    // No error type is provided
    #[enum_str_conv(unknown = unknown_fn)]
    enum MyEnum {
        #[enum_str_conv(str = "apple")]
        A,
    }
}
