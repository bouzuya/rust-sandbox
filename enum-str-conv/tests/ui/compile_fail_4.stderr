error: expected `error` argument: #[enum_str_conv(error = ...)]
 --> tests/ui/compile_fail_4.rs:8:7
  |
8 |     #[enum_str_conv(unknown = unknown_fn)]
  |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
